source/ScreensaverHero.hx

```diff
@@ -0,0 +1,92 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.math.FlxRandom;
+import flixel.util.FlxSpriteUtil;
+
+class ScreensaverHero extends FlxSprite
+{
+	public var direction:Int;
+
+	public function new(x:Int, y:Int)
+	{
+		super(x, y);
+		var randomDirection:FlxRandom = new FlxRandom();
+		this.direction = randomDirection.int(0, 7);
+	}
+
+	override function update(elapsed:Float)
+	{
+		if (this.x < 500)
+		{
+			this.color = 0xff0000;
+		}
+		if (this.x > 500)
+		{
+			this.color = 0x00ff00;
+		}
```

Comment:
Since this code is just focused on changing the color, it's also  a good block to break into its own little private method. Same idea applies to the screen wrapping down below - the more you break up something complex like `update()` the more programmer-friendly it'll be (especially in a collaborative project).